tests/common/type_coverage.h

```diff
@@ -276,6 +420,211 @@ using is_not_a_type_pack =
+  // The first non-pack argument passed into the for_all_combinations stops the
+  // recursion
+  if constexpr (present_in<Needed1, ActionArgsT...> &&
+                present_in<Needed2, ActionArgsT...>)
+    Action<ActionArgsT...>{}(std::forward<HeadT>(head),
+                             std::forward<ArgsT>(args)...);
+}
+
+/**
+ * @brief Overload to handle the iteration over the types within the named type
+ * pack
+ */
+template <template <typename...> class Action, typename Needed1,
+          typename Needed2, typename... ActionArgsT, typename... HeadTypes,
+          typename... ArgsT>
+inline void for_all_combinations_with_two(
+    const named_type_pack<HeadTypes...>& head, ArgsT&&... args) {
+  // Run the next level of recursion for each type from the head named_type_pack
+  // instance. Each recursion level unfolds the first argument passed and adds a
+  // type name as the last argument.
+  size_t type_name_index = 0;
+
+  ((for_all_combinations_with_two<Action, Needed1, Needed2, ActionArgsT...,
+                                  HeadTypes>(std::forward<ArgsT>(args)...,
+                                             head.names[type_name_index]),
+    ++type_name_index),
+   ...);
+  // The unary right fold expression is used for parameter pack expansion.
+  // Every expression with comma operator is strictly sequenced, so we can
+  // increment safely. And of course the fold expression would not be optimized
+  // out due to side-effects.
+  // Additional pair of brackets is required because of precedence of increment
+  // operator relative to the comma operator.
+  //
+  // Note that there is actually no difference in left or right fold expression
+  // for the comma operator, as it would give the same order of actions
+  // execution and the same order of the type name index increment: both the
+  // "(expr0, (exr1, expr2))" and "((expr0, expr1), expr2)" would give the same
+  //  result as simple "expr0, expr1, expr2"
+  assert((type_name_index == sizeof...(HeadTypes)) && "Pack expansion failed");
+}
+
+/**
+ * @brief Overload to handle the iteration over the types within the unnamed
+ * type pack
+ */
+template <template <typename...> class Action, typename Needed1,
+          typename Needed2, typename... ActionArgsT, typename... HeadTypes,
+          typename... ArgsT>
+inline void for_all_combinations_with_two(
+    const unnamed_type_pack<HeadTypes...>& head, ArgsT&&... args) {
+  // Using fold expression to iterate over all types within type pack
+
+  size_t typeNameIndex = 0;
+
+  ((for_all_combinations_with_two<Action, Needed1, Needed2, ActionArgsT...,
+                                  HeadTypes>(std::forward<ArgsT>(args)...),
+    ++typeNameIndex),
+   ...);
+  // Ensure there is no silent miss for coverage
+  assert((typeNameIndex == sizeof...(HeadTypes)) && "Pack expansion failed");
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
```

Comment:
Note: All changes to this file were either moved from accessor_common.h or the removed type_coverage.h.